docs/5.x/spa-tracking.md

```diff
@@ -4,16 +4,29 @@ category: Integrate
 
 # Single-Page Application/Progressive Web App Tracking
 
-Single-page websites and Progressive Web Apps have become a standard over the last years. Getting the tracking of such websites and apps right is crucial to your success as you need to ensure the measured data is meaningful and correct.
+In the rapidly evolving digital landscape, single-page applications (SPAs) and Progressive Web Apps (PWAs) have emerged as a prevalent choice for developers, given their fast load times, smooth user experience, and offline capabilities. This document serves as a comprehensive guide to setting up accurate tracking within SPAs and PWAs using Matomo, an open-source analytics platform. We cover multiple solutions to ensure you gather meaningful and correct data, vital for optimising user experience and achieving your digital objectives.
 
-## If available, consider using an existing integration for your framework
+# Solution 1) Embed Matomo Tag Manager Container Code
 
-If you're using a popular framework such as Angular, Angular JS, Vue.js, React Native, etc. then we recommend to first check if an integration may be available for your framework: [see all available integrations for Matomo](https://matomo.org/integrate/#programming-language-platforms-and-frameworks).
+If you're using [Tag Manager](https://matomo.org/tag-manager/) to implement your Matomo Analytics Tracking, then in your Single Page Application,  whenever there is a new page loaded in your app you will need your [Matomo Tag](https://matomo.org/docs/tag-manager/#configuring-a-tag) to be triggered for the Page view to be tracked. 
+
+To trigger your Matomo tag (which calls `trackPageView`), you can either:
+
+1. use the "History change" [trigger](https://matomo.org/docs/tag-manager/#triggers) which would work in most cases,
```

Comment:
here we should mention about this filter too, else it fires the pageview tag thrice, See #L3-410 for more info
![Screenshot from 2023-02-14 12-23-43](https://github.com/matomo-org/developer-documentation/assets/14902277/a4770e06-e83f-49ea-b085-d5d4ac04ece0)